CONTRIBUTING.md

```diff
@@ -0,0 +1,227 @@
+which is the text to be translated. The second allows for interpolation of values into the translated string.
+
+For example:
+
+```Nunjucks
+<h1>{{ _("Example") }}</h1>
+
+<p>{{ _i("Using ${templateName} template engine.", {templateName: "Nunjucks"}) }}</p>
+```
+
+Would render the follwing in English:
+
+```HTML
+<h1>Example</h1>
+
+<p>Using Nunjucks template engine</p>
+```
+
+A French localization might render like:
+
+```HTML
+<h1>Exemple</h1>
+
+<p>Utiliser le moteur de template Nunjucks</p>
+```
+
+In JavaScript we need to import the [eleventy-plugin-i18n-gettext](https://www.npmjs.com/package/eleventy-plugin-i18n-gettext)
+into our code and make calls to the functions.
+
+```JavaScript
+"use strict";
+
+const i18n = require("eleventy-plugin-i18n-gettext");
+
+module.exports = (data) => {
+    const locale = "en-CA";
+    return i18n._(locale, "To translate");
+};
+```
+
+_**NOTE**: Due to [issue #22](https://github.com/sgissinger/eleventy-plugin-i18n-gettext/issues/22) when passing in the
+locale, it needs to use a variable called `locale`_
+
+_**NOTE**: Due to [issue #23](https://github.com/sgissinger/eleventy-plugin-i18n-gettext/issues/23) when passing in the
+`key`, if it isn't plain text it will not be included in messages.js. A workaround, if a variable is being used, is to
+make a call with the resolved text elsewhere. For examplethe category names, used in URLs, are explicitly called in
+`src/_data/site.js`. This applies to calls made in JavaScript and templates._
+
+The specific localizations are provided in the .po and/or .mo files locataed in the [locales](./src/locales) directory.
+The strings to translate are automatically picked up when a build is generated (`npm start`) and added to
+[messages.js](./src/locales/messages.js). This allows for tools like [Poedit](https://poedit.net) to easily find the
+strings that need to be translated. You could also update the .po and generate the .mo files manually.
+
+#### Using Poedit
+
+You'll need to add a JavaScript extractor when first setting up Poedit. See
+[Poedit configuration for translations extraction](https://github.com/sgissinger/eleventy-plugin-i18n-gettext/blob/HEAD/docs/Manage-translations-with-Poedit.md)
+for information about setup.
+
+To update the .po and .mo files perform the following steps:
+
+1. Launch Poedit
+2. Browse for the .po file you'd like to edit
+3. Open the .po file
+4. `Translations > Update from Source Code`
+   * This will update the .po file with all of the current translations
+   * If this step doesn't pick up the localizations from messages.js, ensure that the source paths are set corrctly.
+     * `Translations > Properties... > Source Paths
+5. Select the "Source Text" to localize, add the necessary "Translation"
+6. To remove old strings go to: `Translations > Purge Deleted Translations`
+7. Save
+8. Make sure to commit the updates .po and .mo files to the repo.
```

Comment:
```suggestion
To update the `.po` and `.mo` files perform the following steps:

1. Launch Poedit.
2. Browse and open the `.po` file to edit.
3. Select `Translations > Update from Source Code`.
   * This will update the `.po` file with all of the current translations.
   * If this step does not find the available localizations from the `./src/locales/messages.js` file, ensure that the source paths are set correctly: `Translations > Properties... > Source Paths`.
5. Select the "Source Text" to localize, add the necessary "Translation".
6. To remove old strings go to: `Translations > Purge Deleted Translations`.
7. Save.
8. Make sure to commit the updates `.po` and `.mo` files to the working repository.
```